src/stylesheets/animations.less

```diff
@@ -1,209 +1,207 @@
+        }
 
-@-webkit-keyframes sk-appear-frames-md {
-    0% {
-        bottom: @widget-close-bottom-md;
-    }
-    100% {
-        bottom: @widget-vertical-spacing;
-    }
-}
-@keyframes sk-appear-frames-md {
-    0% {
-        bottom: @widget-close-bottom-md;
-    }
-    100% {
-        bottom: @widget-vertical-spacing;
-    }
-}
+        @media (min-width: @screen-sm-min) and (max-height: @screen-xs-ht-max) {
+            bottom: -@widget-height-sm;
+        }
 
-@-webkit-keyframes sk-appear-frames-lg {
-    0% {
-        bottom: @widget-close-bottom-lg;
-    }
-    100% {
-        bottom: @widget-vertical-spacing;
-    }
-}
-@keyframes sk-appear-frames-lg {
-    0% {
-        bottom: @widget-close-bottom-lg;
-    }
-    100% {
-        bottom: @widget-vertical-spacing;
+        // Mobiles
+        @media (max-width: @screen-sm-min) {
+            bottom: -@widget-height-xs;
+        }
     }
-}
 
-@-webkit-keyframes sk-appear-frames-sm {
-    0% {
-        bottom: @widget-close-bottom-sm;
+    @keyframes sk-appear-badge-frames {
+        0% {
+            opacity: 0;
+            transform: scale(0, 0);
+        }
+        100% {
+            opacity: 1;
+            transform: scale(1, 1);
+        }
     }
-    100% {
-        bottom: @widget-vertical-spacing;
+
+    @keyframes sk-close-badge-frames {
+        0% {
+            opacity: 1;
+            transform: scale(1, 1);
+        }
+        100% {
+            opacity: 0;
+            transform: scale(0, 0);
+        }
     }
 }
-@keyframes sk-appear-frames-sm {
-    0% {
-        bottom: @widget-close-bottom-sm;
-    }
-    100% {
+
+.sk-tab-display {
+    &.sk-appear {
         bottom: @widget-vertical-spacing;
-    }
-}
 
-@-webkit-keyframes sk-appear-frames-xs {
-    0% {
-        bottom: @widget-close-bottom-xs;
-    }
-    100% {
-        bottom: 0;
-    }
-}
-@keyframes sk-appear-frames-xs {
-    0% {
-        bottom: @widget-close-bottom-xs;
-    }
-    100% {
-        bottom: 0;
+        animation: sk-appear-tab-frames-md .4s cubic-bezier(.62, .28, .23, .99);
+        animation-delay: .0s;
+        animation-fill-mode: forwards;
+
+        @media (min-width: @screen-lg-min) and (min-height: @screen-md-ht-min) {
+            animation: sk-appear-tab-frames-lg .4s cubic-bezier(.62, .28, .23, .99);
+        }
+
+        @media (min-width: @screen-sm-min) and (max-height: @screen-sm-ht-max) {
+            animation: sk-appear-tab-frames-sm .4s cubic-bezier(.62, .28, .23, .99);
+        }
+
+        // Mobile
+        @media (max-width: @screen-xs-max) {
+            bottom: 0;
+            animation: sk-appear-tab-frames-xs .4s cubic-bezier(.62, .28, .23, .99);
+        }
     }
-}
 
-.sk-close {
-    bottom: @widget-close-bottom-md;
+    &.sk-close {
+        bottom: @widget-close-bottom-md;
 
-    -webkit-animation: sk-close-frames-md .4s cubic-bezier(.62, .28, .23, .99);
             animation: sk-close-frames-md .4s cubic-bezier(.62, .28, .23, .99);
```

Comment:
Are all the `sk-close-frames` related to the tab widget? If so, I would rename to `sk-close-tab-frames` to be consistent with the `sk-close-badge-frames`